### DISCUSSION TITLE
Using ffmpeg

### DISCUSSION THREAD

**Discussion by Jay123210599** [staxrip/staxrip#1295] [Q&A]
Does it already use ffmpeg to make gifs, apngs, and animated webps? If so, if I split a video into multiple selection and want to make animated stills out of all of them, how do I do it without getting this?
![Screenshot (421)](https://github.com/staxrip/staxrip/assets/141977822/651f7c93-38c9-4693-a83f-154a29070f5a)

But if not, how do I enter ffmpeg commands into Staxrip? How do I batch process files using ffmpeg? What is the difference between the regular H.264 & H.265 encoders and the ones from ffmpeg? How do I get lossless from them?

**Comment by jessielw** [comment]
This is a lot of questions in one. 

You realistically should only use FFMPEG to use FFMPEG. It is very powerful, but when using a frontend like StaxRip or even your own scripts you should use AviSynth or VapourSynth pushed to x264/x265. 

FFMPEG works very similar and can provide the same quality as them, but they are the official encoders for the codec to my understanding. 

I'm not sure if StaxRip provides the features you want in the UI to make animated stills. This is likely something you'll need to learn to script with VapourSynth/AviSynth or potentially can do some of it via CLI with FFMPEG. Although, you're likely to get better results using the scripting languages.
